presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveTransactionalTable.java

```diff
@@ -355,6 +357,64 @@ public void testCreateAcidTable(boolean isPartitioned, BucketingType bucketingTy
         }
     }
 
+    @Test(dataProvider = "isTablePartitioned", groups = HIVE_TRANSACTIONAL)
+    public void testFilesForAbortedTransactionsIgnored(boolean isPartitioned)
+            throws TException
+    {
+        if (getHiveVersionMajor() < 3) {
+            throw new SkipException("Hive transactional tables are supported with Hive version 3 or above");
+        }
+
+        String tableName = "test_aborted_transaction_table_partitioned" + (isPartitioned ? "_partitioned" : "");
+        onHive().executeQuery("" +
+                "CREATE TABLE " + tableName + " (col INT) " +
+                (isPartitioned ? "PARTITIONED BY (part_col INT) " : "") +
+                "STORED AS ORC " + "TBLPROPERTIES ('transactional'='true')");
+
+        try {
+            String hivePartition = isPartitioned ? " PARTITION (part_col=2) " : "";
+            String predicate = isPartitioned ? " WHERE part_col = 2 " : "";
+
+            onHive().executeQuery("INSERT OVERWRITE TABLE " + tableName + hivePartition + " select 1");
+
+            String selectFromOnePartitionsSql = "SELECT col FROM " + tableName + predicate + " ORDER BY COL";
+            QueryResult onePartitionQueryResult = query(selectFromOnePartitionsSql);
+            assertThat(onePartitionQueryResult).containsOnly(row(1));
+
+            onHive().executeQuery("INSERT INTO TABLE " + tableName + hivePartition + " select 2");
+            onePartitionQueryResult = query(selectFromOnePartitionsSql);
+            assertThat(onePartitionQueryResult).containsExactly(row(1), row(2));
+
+            onHive().executeQuery("INSERT OVERWRITE TABLE " + tableName + hivePartition + " select 3");
+            onePartitionQueryResult = query(selectFromOnePartitionsSql);
+            assertThat(onePartitionQueryResult).containsOnly(row(3));
+
+            onHive().executeQuery("INSERT INTO TABLE " + tableName + hivePartition + " select 4");
+            onePartitionQueryResult = query(selectFromOnePartitionsSql);
+            assertThat(onePartitionQueryResult).containsExactly(row(3), row(4));
```

Comment:
Do we need that many inserts before actually simulating aborted transaction.
It is not obvious what cases are we testing here.

Reply:
yeah, have removed some of them.

Reply:
thanks